checks.d/mongo.py

```diff
@@ -354,49 +352,52 @@ def __init__(self, name, init_config, agentConfig, instances=None):
     def get_library_versions(self):
         return {"pymongo": pymongo.version}
 
-    def check_last_state(self, state, clean_server_name, agentConfig):
-        if self._last_state_by_server.get(clean_server_name, -1) != state:
-            self._last_state_by_server[clean_server_name] = state
-            return self.create_event(state, clean_server_name, agentConfig)
-
-    def create_event(self, state, clean_server_name, agentConfig):
+    def check_last_state(self, state, clean_server_name, replSet, agentConfig):
+        last_state = self._last_state_by_server.get(clean_server_name, -1)
+        self._last_state_by_server[clean_server_name] = state
+        if last_state != state and last_state != -1:
+            return self.create_event(last_state, state, clean_server_name, replSet['set'], agentConfig)
+
+    def hostname_for_event(self, clean_server_name, agentConfig):
+        """Return a reasonable hostname for a replset membership event to mention."""
+        uri = urlsplit(clean_server_name)
+        hostname = uri.netloc.split(':')[0]
+        if hostname == 'localhost':
+            hostname = get_hostname(agentConfig)
+        return hostname
+
+    def create_event(self, last_state, state, clean_server_name, replset_name, agentConfig):
         """Create an event with a message describing the replication
             state of a mongo node"""
 
         def get_state_description(state):
-            if state == 0:
-                return 'Starting Up'
-            elif state == 1:
-                return 'Primary'
-            elif state == 2:
-                return 'Secondary'
-            elif state == 3:
-                return 'Recovering'
-            elif state == 4:
-                return 'Fatal'
-            elif state == 5:
-                return 'Starting up (forking threads)'
-            elif state == 6:
-                return 'Unknown'
-            elif state == 7:
-                return 'Arbiter'
-            elif state == 8:
-                return 'Down'
-            elif state == 9:
-                return 'Rollback'
+            return self.REPLSET_MEMBER_STATES[state][1]
```

Comment:
As for the `get_state_name ` method, can we consider the case where `state` does not exist in `REPLSET_MEMBER_STATES ` ?
``` python
if state in 
  pass
else:
  return 'UNKNOWN'
```

Even though, this should not happen (as opposed to `get_state_name` when `last_state==-1` ), I think that would make the check less error prone to MongoDB updates.